pkg/file/xar_test.go

```diff
@@ -0,0 +1,61 @@
+package file
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCheckPKGSignature(t *testing.T) {
+	testCases := []struct {
+		in  []byte
+		out error
+	}{
+		{in: []byte{}, out: io.EOF},
+		{
+			in:  tarFile,
+			out: ErrInvalidType,
+		},
+		{
+			in:  unsignedFile,
+			out: ErrNotSigned,
+		},
+		{
+			in:  signedFile,
+			out: nil,
+		},
+		{
+			in:  wrongTOCFile,
+			out: errors.New("decompressing TOC: unexpected EOF"),
+		},
+	}
+
+	for _, c := range testCases {
+		r := bytes.NewReader(c.in)
+		err := CheckPKGSignature(r)
+		if c.out != nil {
+			require.ErrorContains(t, err, c.out.Error())
+		} else {
+			require.NoError(t, err)
+		}
+	}
+}
+
+// TODO: it would be ideal if we could generate these dinamically instead of
+// hardcoding blobs.
+//
+// To generate the following files, I printed into stdout something on the
+// lines of response.body[:header.CompressedSize + 28] (you can get this value from
+// parseHeader)
+var (
```

Comment:
Just a suggestion, it may be a bit nicer to store those as binary files under a `testdata` directory? We could even have scripts (in `tools/` maybe?) to re-generate them should it be needed.

Reply:
I like that, would it be good enough if I move them to a `testdata` directory and follow up with the auto-generation once this feature is done?

Reply:
For sure! Or even only the first time it is needed, we may never need to regenerate that again AIUI.